autograder/rest_api/views/group_views.py

```diff
@@ -382,3 +385,65 @@ def _get_merged_extended_due_date(self, group1, group2):
             return group1.extended_due_date
 
         return max(group1.extended_due_date, group2.extended_due_date)
+
+
+def clean_extended_due_dates(update_data: dict, old_group: ag_models.Group) -> dict:
+    """
+    Return a new dict without 'soft_extended_due_date' and without 'hard_extended_due_date'
+    if `update_data` contains a changed value for 'extended_due_date'. If it contains just
+    a changed value to either 'soft_extended_due_date' or 'hard_extended_due_date', return
+    a new dict without 'extended_due_date'.
+
+    :raises: exceptions.ValidationError if `update_data` contains new values for
+             both deprecated 'extended_due_date' and one of 'soft_extended_due_date' or
+             'hard_extended_due_date', or if any datetime strings in `update_data` are invalid.
+    """
+    legacy_changed = current_changed = False
+    try:
+        if ('extended_due_date' in update_data and not core_ut.datetimes_are_equal(
```

Comment:
Change to `legacy_changed = condition` (avoids "if condition, value = True, else value = False" pattern)